src/hardware/vga_other.cpp

```diff
@@ -122,8 +129,10 @@ static Bitu read_crtc_data_other(Bitu /*port*/,Bitu /*iolen*/) {
 		return vga.other.hsyncp;
 	case 0x03:		//Horizontal and vertical sync width
 		if (machine==MCH_TANDY)
-			return vga.other.hsyncw | (vga.other.vsyncw << 4);
-		else return vga.other.hsyncw;
+			return static_cast<uint8_t>(vga.other.hsyncw |
+			                            (vga.other.vsyncw << 4));
```

Comment:
This bit operation is… weird… Are we packing 2 4-bit values into a single 8-bit value in here? Perhaps we should add some runtime asserts?

Reply:
Looks like it! The only place `hsyncw` is assigned is indeed 4-bits:

``` c++
vga.other.hsyncw = val & 0xf;
```

> Perhaps we should add some runtime asserts?

:+1:

Reply:
Fixup inbound.